lib/sycamore/sycamore/reader.py

```diff
@@ -217,6 +217,7 @@ def opensearch(
         index_name: str,
         query: Optional[Dict] = None,
         reconstruct_document: bool = False,
+        query_kwargs: dict[str, Any] = {},
```

Comment:
I'm a bit unsure why we would keep `kwargs` at all here since it isn't being used. It could be quite confusing to customers. Is there no way to filter `kwargs` to only pass valid values in? That might be an easier fix.